rclpy/src/rclpy/_rclpy.c

```diff
@@ -5494,6 +5506,62 @@ rclpy_publisher_get_topic_name(PyObject * Py_UNUSED(self), PyObject * args)
   return PyUnicode_FromString(topic_name);
 }
 
+
+/// Retrieves default domain id defined in rcl layer
+/**
+ *
+ * \return the defined domain id
+ */
+static PyObject *
+rclpy_get_default_domain_id(PyObject * Py_UNUSED(self), PyObject * Py_UNUSED(args))
+{
+  return PyLong_FromSize_t(RCL_DEFAULT_DOMAIN_ID);
+}
+
+/// Retrieves domain id from init_options of context
+/**
+ * \param[in] pyargs context Python object
+ * \return domain id
+ */
+static PyObject *
+rclpy_get_domain_id_in_context(PyObject * Py_UNUSED(self), PyObject * args)
```

Comment:
It would be more accurate to use here https://github.com/ros2/rcl/blob/70d13257a97f1278fd8e8a38f380e33b011f32e7/rcl/include/rcl/context.h#L267, which is the actual domain id used by the context.

Reply:
@Barry-Xu-2018 this comment still seems to be valid.